rabbitmq/producer.go

```diff
@@ -0,0 +1,62 @@
+package rabbitmq
+
+import (
+        "fmt"
+
+        "github.com/streadway/amqp"
+	    log "github.com/sirupsen/logrus"
+)
+
+type RabbitMQOptions struct {
+	Host                string
+	Port                int64
+	ExchangeName        string
+	ExchangeRoutingKey  string
+}
+
+func failOnError(err error, msg string) {
+        if err != nil {
+                log.Fatalf("%s: %s", msg, err)
+        }
+}
+
+func Setup(options RabbitMQOptions) (*amqp.Connection, *amqp.Channel){
+  // 1. Create the connection to RabbitMQ
+  url := fmt.Sprintf("amqp://guest:guest@%s:%d", options.Host, options.Port)
```

Comment:
Extract into function `func (options RabbitMQOptions) ConnectionURI() string` that provides this

Reply:
Fixed